### PR TITLE
[DOCS] Add TOC to landing page

### PR DESCRIPTION
**Pull request by jrodewig**
### Summary
- Adds the TOC to the Elasticsearch docs landing page. Removes the right sidebar from the landing page.
- Removes the "View all Elastic docs" link from the bottom of the landing page

### Screenshots

<details><summary><b>Before</b></summary>

elasticsearch-before

</details> 

<details><summary><b>After</b></summary>

elasticsearch-after

</details>

### CODE REVIEW

File: docs/reference/landing-page.asciidoc [hunk 1]
```diff
@@ -240,4 +241,19 @@ <h4 class="mt-3">
   </div>
 </div>
 
-<p class="my-4"><a href="https://www.elastic.co/guide/index.html">View all Elastic docs</a></p>
+<script>
```
Comment (jrodewig) [inline docs/reference/landing-page.asciidoc:244]: This script was copied from the Observability docs:
https://github.com/elastic/observability-docs/blob/2c5ec189bd08f43366a39796137ad5c26c139247/docs/en/observability/landing-page/temp/script.asciidoc?plain=1#L2

File: docs/reference/landing-page.asciidoc [hunk 2]
```diff
@@ -240,4 +241,19 @@ <h4 class="mt-3">
...
+window.addEventListener("DOMContentLoaded", (event) => {
+  const left_col = document.getElementById("left_col")
+  left_col.classList.remove('col-0')
+  left_col.classList.add("col-12", "col-md-4", "col-lg-3", "h-almost-full-md", "sticky-top-md")
```
Comment (abdonpijpelink) [inline docs/reference/landing-page.asciidoc:248]:
```suggestion
  left_col.classList.add("col-12", "order-2", "col-md-4", "col-lg-3", "h-almost-full-md", "sticky-top-md")
```

Comment (abdonpijpelink) [reply docs/reference/landing-page.asciidoc:248]: Maybe this was the intention, but on a phone, the TOC is rendered before the content (contrary to the other pages, where the TOC is rendered after the content). I personally like after better, which we can do by adding this CSS class to the left column.

**Review by abdonpijpelink — APPROVED** [review]
LGTM. This is great @jrodewig . Thank you.

I've left a small suggestion.

File: docs/reference/landing-page.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/landing-page.asciidoc:248]: Good feedback. Thanks!

Comment (jrodewig) [reply docs/reference/landing-page.asciidoc]: I ended up reverting this change with https://github.com/elastic/elasticsearch/commit/52a915bf31eabefc46d1a8e2c7e2c6f8ca82c76c.

It looks like the class also changes the order on larger screens. This seems like a relatively minor detail so I don't think it's worth addressing right now.

**Comment by jrodewig** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.9|[](https://github.com/elastic/elasticsearch/pull/97447)|
|✅|8.8|[](https://github.com/elastic/elasticsearch/pull/97448)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**